sdk/servicebus/service-bus/src/serializers/queueResourceSerializer.ts

```diff
@@ -0,0 +1,295 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+import { AtomXmlSerializer, HttpOperationResponse, XMLRequestInJSON } from "@azure/core-http";
+import * as Constants from "../util/constants";
+import { serializeToAtomXmlRequest, deserializeAtomXmlResponse } from "../util/atomXmlHelper";
+import {
+  getStringOrUndefined,
+  getIntegerOrUndefined,
+  getBooleanOrUndefined,
+  getCountDetailsOrUndefined,
+  CountDetails
+} from "../util/utils";
+
+const requestProperties: Array<keyof InternalQueueOptions> = [
+  Constants.LOCK_DURATION,
```

Comment:
@ramya-rao-a I recall you had questions/concerns about usage of the constants Vs directly using the string ("LockDuration") in this case.
There a specific one we want to follow? @chradek

Reply:
Since the strings are used in multiple places, going with using the constants approach.

About the PR
- Added in more changes based on findings as noted in https://github.com/Azure/azure-sdk-for-js/issues/5354#issuecomment-538076734
For correctness and support of the individual properties, best effort changes are added in based on what was being supported in azure-sb. For cross language parity and detailed checks on the properties (basically more thorough testing, this is a WIP and depends on things that come up as part of the investigation).
We could potentially hold off on merging PRs to master till the issues are resolved, or merge in if we are okay with what is usable with current set of changes. The code in working state so far is checked in on feature branch temporarily for reference - https://github.com/Azure/azure-sdk-for-js/tree/feature/service-bus-atom-mgmt-api-prv/sdk/servicebus/service-bus/src (in case we want to continue prioritizing issues before shipping it and unblock other progress on master, such as other ongoing Service Bus bug fixes)

cc @ramya-rao-a @AlexGhiondea @chradek